### PR TITLE
Update tools, structure and Kubernetes version

### LINKED ISSUE #526: Tests are missing proper mock checks
### What is missing?

A lot of tests have mocks used, but their assertions are never called - thus not verifying if the calls were actually made or not. This makes a lot of tests a bit questionable if they're actually testing what they're supposed to.

### Why is this needed?

Tests should be trusted to verify that they test what they claim to test.

### LINKED ISSUE #531: Update to Kubebuilder gov4 layout
### What is missing?

Kubebuilder is moving to another directory layout structure. This hides functionality that's not intended to be exported to internal directory (such as controllers). 

### Why is this needed?

Maintenance.

### PR DESCRIPTION
**Pull request by burmanm**
**What this PR does**:
This is a subset of PR #524 , removing Kustomize v5 template changes

* Updates envtest base Kubernetes to 1.26
* Updates controller-runtime
* Fixes the flaky test
* Updates project structure to gov4-alpha, move controllers to different directory
* Regenerates and updates Mocks
* Improve testing quality by actually verifying all the Mocked method calls are done (and nothing extra)
* Adds more arch targets to Dockerfile

**Which issue(s) this PR fixes**:
Fixes #526
Fixes #531

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: pkg/mocks/helper.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	// var r0 client.SubResourceWriter
+	// if rf, ok := ret.Get(0).(func() client.SubResourceWriter); ok {
+	// 	r0 = rf()
+	// } else {
+	// 	if ret.Get(0) != nil {
+	// 		r0 = ret.Get(0).(client.SubResourceWriter)
+	// 	}
+	// }
+	// return r0
```
Comment (adejanovski) [inline pkg/mocks/helper.go]:
```suggestion
```

Could we get rid of the commented lines if they're not used?

File: pkg/mocks/helper.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	// var r0 client.SubResourceClient
+	// if rf, ok := ret.Get(0).(func(string) client.SubResourceClient); ok {
+	// 	r0 = rf(subResource)
+	// } else {
+	// 	if ret.Get(0) != nil {
+	// 		r0 = ret.Get(0).(client.SubResourceClient)
+	// 	}
+	// }
+
+	// return r0
```
Comment (adejanovski) [inline pkg/mocks/helper.go]:
```suggestion
```

Same question here

**Review by adejanovski — CHANGES_REQUESTED** [review]
I have a small request for removing commented code.
Other than that, just waiting for failed checks to pass after re-running them.

File: pkg/mocks/helper.go (hunk 2 above)
Comment (burmanm) [reply pkg/mocks/helper.go]: Done

**Review by adejanovski — APPROVED** [review]

**Pull request merged**